File: BLOCK1/LAB1_2/lab 1 2.cpp
#include <iostream>
using namespace std;

int main()
{
    setlocale(LC_ALL, "Russian");
    const int LIMIT{ 12 };
    int n = 0, m = 0, placements = 0;
    bool isIncorrect;

    cout << "Данная программа вычисляет число размещений из N по M." << endl;
    do {
        isIncorrect = false;
        cout << "Условия ввода: N и M натуральные числа, N больше M, N и M меньше 12." << endl;
        cout << "Введите N: ";
        cin >> n;
        if (cin.get() != '\n') {
            cin.clear();
            while (cin.get() != '\n');
            cout << "Введенные данные не соответствуют условию. Повторите попытку." << endl;
            isIncorrect = true;
        }
        else {
            cout << "Введите M: ";
            cin >> m;
        }
        if (!isIncorrect && cin.get() != '\n') {
            cin.clear();
            while (cin.get() != '\n');
            cout << "Введенные данные не соответствуют условию. Повторите попытку." << endl;
            isIncorrect = true;
        }
        if (!isIncorrect &&
            ((n < m + 1) || (n < 0) || (m < 0) || (n > LIMIT) || (m > LIMIT))) {
            cout << "Введенные данные не соответствуют условию. Повторите попытку." << endl;
            isIncorrect = true;
        }
    } while (isIncorrect);
    placements = n;

    int temp = n - m;
    
    for (int i = n - 1; i > temp; i--)
    {
        n--;
        placements = placements * n;
    }

    cout << "Число размещений равно " << placements << endl;

    return 0;
}

File: BLOCK1/LAB1_2/lab12.java
import java.util.Scanner;

public class lab12{

    public static void main(String[] args){

        Scanner scan=new Scanner(System.in);
        final int LIMIT = 12;
        int n = 0, m = 0, placements = 0;
        boolean isIncorrect;

        System.out.println ("Данная программа вычисляет число размещений из N по M.");
        do {
            isIncorrect = false;
            System.out.println("Условия ввода: N и M натуральные числа, N больше M, N и M меньше 12.");
            try {
                System.out.print("Введите N: ");
                n = Integer.parseInt(scan.nextLine());
                System.out.print("Введите M: ");
                m = Integer.parseInt(scan.nextLine());
            }
            catch (NumberFormatException e) {
                System.out.println("Введенные данные не соответствуют условию. Повторите попытку.");
                isIncorrect = true;
            }
            if (!isIncorrect && ((n < m + 1) || (n < 1) || (m < 1) || (n > LIMIT) || (m > LIMIT))) {
                System.out.println("Введенные данные не соответствуют условию. Повторите попытку.");
                isIncorrect = true;
            }
            } while (isIncorrect);
        scan.close();
        placements = n;
        int temp = n - m;

        for (int i = n - 1; i >temp; i--)
        {
            n--;
            placements = placements * n;
        }
        System.out.println ("Число размещений равно " + placements);
    }
}